R/generateThreshVsPerf.R

```diff
@@ -388,28 +392,29 @@ plotROCCurves = function(obj, measures, diagonal = TRUE, pretty.names = TRUE) {
     mlearn = FALSE
   resamp = "iter" %in% colnames(obj$data)
 
-  if (!obj$aggregate & mlearn & resamp) {
+
+  aes = list(x = names(measures)[1], y = names(measures)[2])
+
+  if (!obj$aggregate && mlearn && resamp) {
     obj$data$int = interaction(obj$data$learner, obj$data$iter)
-    p = ggplot(obj$data, aes_string(names(measures)[1], names(measures)[2], group = "int"))
-    p = p + geom_path(alpha = .5)
-  } else if (!obj$aggregate & !mlearn & resamp) {
-    p = ggplot(obj$data, aes_string(names(measures)[1], names(measures)[2], group = "iter"))
-    p = p + geom_path(alpha = .5)
-  } else if (obj$aggregate & mlearn & !resamp) {
-    p = ggplot(obj$data, aes_string(names(measures)[1], names(measures)[2]), group = "learner", color = "learner")
-    p = p + geom_path(alpha = .5)
+    aes$group = "int"
+  } else if (!obj$aggregate && !mlearn && resamp) {
+    aes$group = "iter"
+  } else if (obj$aggregate && mlearn && !resamp) {
+    aes$group = "learner"
   } else {
     obj$data = obj$data[order(obj$data$threshold), ]
-    p = ggplot(obj$data, aes_string(names(measures)[1], names(measures)[2]))
-    p = p + geom_path()
   }
 
-  p = p + labs(x = mnames[1], y = mnames[2])
+  if (mlearn && !facet.learner)
+    aes$color = "learner"
+
+  p = ggplot(obj$data, do.call(aes_string, aes)) + geom_path() + labs(x = mnames[1], y = mnames[2])
 
-  if (length(unique(obj$data$learner)) > 1L)
+  if (mlearn && facet.learner)
     p = p + facet_wrap(~ learner)
 
-  if (diagonal & all(sapply(obj$data[, names(measures)], function(x) max(x, na.rm = TRUE)) <= 1))
+  if (diagonal && all(sapply(obj$data[, names(measures)], function(x) max(x, na.rm = TRUE)) <= 1))
```

Comment:
Oh no, there's an `sapply` -- didn't we decide we don't want those anymore? (I know that it was there before, but since you're touching this code anyway, could you change it please?)